src/transformers/benchmark/cache_benchmark.py

```diff
@@ -0,0 +1,180 @@
+
+    def target(self, batch_size, max_cache_length, seq_length, cache_type, mode):
+        self.model.generation_config.max_length = max_cache_length
+        input_ids = (
+            self.tokenizer([self.input_text] * batch_size, return_tensors="pt")
+            .to(self.device)
+            .input_ids[:batch_size, :seq_length]
+        )
+
+        self.model._reset_cache()
```

Comment:
no all model support this, same for setup cache but you might be error out earlier I'll keep reading